pootle/static/js/editor/app.js

```diff
@@ -1749,57 +1746,58 @@ PTL.editor = {
       return false;
     }
 
-    const filterChecks = this.$filterChecks.val();
+    const name = this.$filterChecks.val();
+    const isCategory = $(this.$filterChecks.select2('data').element)
+      .data('type') === 'category';
 
-    if (filterChecks !== 'none') {
-      const sortBy = this.$filterSortBy.val();
-      const newHash = {
-        filter: 'checks',
-        checks: filterChecks,
-      };
+    const sortBy = this.$filterSortBy.val();
+    const newHash = {
+      filter: 'checks',
+    };
 
-      if (sortBy !== 'default') {
-        newHash.sort = sortBy;
+    if (name !== 'all') {
+      if (isCategory) {
+        newHash.category = name;
+      } else {
+        newHash.checks = name;
       }
+    }
 
-      $.history.load($.param(newHash));
+    if (sortBy !== 'default') {
+      newHash.sort = sortBy;
     }
+
+    $.history.load($.param(newHash));
     return true;
   },
 
   /* Adds the failing checks to the UI */
-  appendChecks(checks) {
-    if (Object.keys(checks).length) {
-      const $checks = this.$filterChecks;
-      const selectedValue = this.checks[0] || 'none';
-
-      $checks.find('optgroup').each(function displayGroups() {
-        const $gr = $(this);
-        let empty = true;
-
-        $gr.find('option').each(function displayOptions() {
-          const $opt = $(this);
-          const value = $opt.val();
-
-          if (value in checks) {
-            empty = false;
-            $opt.text(`${$opt.data('title')}(${checks[value]})`);
-          } else {
-            $opt.remove();
-          }
-        });
+  updateChecksDropdown(checks) {
+    const $checks = this.$filterChecks;
+
+    $checks.find('optgroup').each(function displayGroups() {
+      const $gr = $(this);
+      let groupTotal = 0;
+
+      $gr.find('option').each(function displayOptions() {
+        const $opt = $(this);
+        const value = $opt.val();
+
+        if ($opt.hasClass('category')) {
+          return;
+        }
 
-        if (empty) {
-          $gr.hide();
+        if (value in checks) {
+          groupTotal += checks[value];
+        } else {
+          $opt.remove();
         }
       });
 
-      $checks.select2(filterSelectOpts).select2('val', selectedValue);
-      this.$filterChecksWrapper.css('display', 'inline-block');
-    } else { // No results
-      this.displayMsg({ body: gettext('No results.') });
-      this.$filterStatus.select2('val', this.filter);
-    }
+      if (groupTotal === 0) {
+        $gr.hide();
```

Comment:
This is also a similar case as above (indentation level decreased), but we need to be aware that `hide()` on `optgroup`s doesn't work reliably across browsers.